perllib/FixMyStreet/App/Controller/Report/New.pm

```diff
@@ -1046,7 +1046,13 @@ sub process_report : Private {
     $report->detail( $detail );
 
     # mobile device type
-    $report->service( $params{service} ) if $params{service};
+    if ($c->get_param('submit_register')) {
```

Comment:
Small point, but should we handle these via the `$params` hash, as we do for everything else at this point?

Reply:
You'll see that, in the main, params is for all the variables, and get_param is used for submit buttons (e.g. report_anonymously, submit_sign_in bit above). Don't think it's totally consistent, but whatever :)